### PR TITLE
Simplify type parameters for schedule_common()

### PR DESCRIPTION
**Pull request by qinsoon**
This PR
* adds a new trait `GCWorkContext` which provides a few associated types used for work packets. Each plan provides an implementation of `GCWorkContext`.
* moves `schedule_common` from `CommonPlan` to `GCWorkScheduler`.
* fixes tutorial code for the changes in https://github.com/mmtk/mmtk-core/pull/471.

### CODE REVIEW

File: src/plan/global.rs [hunk 1]
```diff
@@ -920,29 +919,27 @@ impl<VM: VMBinding> CommonPlan<VM> {
...
-    pub fn schedule_common<
-        P: Plan<VM = VM>,
-        E: ProcessEdgesWork<VM = VM>,
-        C: CopyContext<VM = VM> + GCWorkerLocal,
-    >(
+    pub fn schedule_common<C: GCWorkContext<VM = VM> + 'static>(
         &self,
-        plan: &'static P,
+        plan: &'static C::PlanType,
         constraints: &'static PlanConstraints,
```
[the remark this answers is not in the document]
Comment (qinsoon) [reply src/plan/global.rs]: Done.

File: src/plan/global.rs [hunk 2]
```diff
@@ -920,29 +919,27 @@ impl<VM: VMBinding> CommonPlan<VM> {
...
         scheduler: &GCWorkScheduler<VM>,
```
[the remark this answers is not in the document]
Comment (qinsoon) [reply src/plan/global.rs]: Good point. I have moved the method to `GCWorkScheduler`. I didn't want to make it the default implementation for the `Plan` trait, as the default implementation cannot be reused if a plan decide to override it.

File: docs/tutorial/src/mygc/ss/collection.md [hunk 3]
```diff
@@ -132,6 +132,17 @@ scheduler's prepare stage and resumes the mutators. The `StopMutators` work
 will invoke code from the bindings to scan threads and other roots, and those 
 scanning work will further push work for a transitive closure.
 
+Though you can add those work packets by yourself, `CommonPlan` provides a
```
Comment (wks) [inline docs/tutorial/src/mygc/ss/collection.md:135]: It is now `Scheduler.schedule_common_work`.  There are several other places mentioning `schedule_common` in this file.

File: src/scheduler/work.rs [hunk 4]
```diff
@@ -22,3 +22,19 @@ pub trait GCWork<VM: VMBinding>: 'static + Send {
         stat.end_of_work(&mut worker.stat);
     }
 }
+
+use super::gc_work::ProcessEdgesWork;
+use crate::plan::CopyContext;
+use crate::plan::Plan;
+
+/// This trait provides a group of associated types that are needed to
+/// create GC work packets for a certain plan. For example, `CommonPlan.schedule_common()`
```
Comment (wks) [inline src/scheduler/work.rs:31]: It is now `Scheduler.schedule_common_work`.

**Review by wks — CHANGES_REQUESTED** [review]
There are a few comments that need updating.

**Comment by qinsoon** [comment]
I have addressed the reviews. @wks

**Review by wks — APPROVED** [review]
LGTM

**Pull request merged**